### PR TITLE
Magento_Directory: avoid using deprecated escape* methods from Abstra…

### PR DESCRIPTION
**Pull request by sergiynezbritskiy**
### Description (*)

Avoid using deprecated escape* methods from AbstractBlock

### Related Pull Requests

https://github.com/magento/magento2/pull/31610

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
@magento run all tests

File: app/code/Magento/Directory/view/adminhtml/templates/js/optional_zip_countries.phtml [hunk 1]
```diff
@@ -12,6 +12,7 @@
 
 /**
  * @var \Magento\Backend\Block\Template $block
+ * @var \Magento\Framework\Escaper $escaper
  * @var \Magento\Framework\View\Helper\SecureHtmlRenderer $secureRenderer
```
Comment (ihor-sviziev) [inline app/code/Magento/Directory/view/adminhtml/templates/js/optional_zip_countries.phtml:14]: please combine all phpdoc blocks to a single one and remove not needed closing and opening PHP tags

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: app/code/Magento/Directory/view/adminhtml/templates/js/optional_zip_countries.phtml (hunk 1 above)
Comment (sergiynezbritskiy) [reply app/code/Magento/Directory/view/adminhtml/templates/js/optional_zip_countries.phtml:14]: fixed

**Comment by sergeynezbritskiy** [comment]
@magento run Functional Tests EE, Integration Tests, Static Tests

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE, Integration Tests

**Review by ihor-sviziev — APPROVED** [review]
✔ Approved.

Failing tests looks not related to changes form this PR.

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Pull request closed without merging**